spec/formatting.md

```diff
@@ -466,6 +468,10 @@ according to their _key_ values and selecting the first one.
 > because no matching _variant_ is available.
 
 The number of _keys_ in each _variant_ MUST equal the number of _selectors_.
+If it does not, emit a _Variant Key Mismatch_ error
```

Comment:
This text also seems to written to allow the implementation to choose whether to recover from data model errors (an implementation that checks for them statically would never reach this point and would just use the fallback value for the result of formatting, and then this would be an assertion rather than a dynamic error).

Which is also fine, but reinforces my suggestion to open a separate issue to discuss that.

Reply:
I created #735 so that, if this PR lands, we don't forget to discuss data model errors.